dashboard/test/controllers/code_reviews_controller_test.rb

```diff
@@ -0,0 +1,98 @@
+require 'test_helper'
+
+class CodeReviewsControllerTest < ActionController::TestCase
+  self.use_transactional_test_case = true
+
+  setup_all do
+    @project_owner = create :student
+    @storage_id = create_storage_id_for_user(@project_owner.id)
+    @channel_id = create :project, storage_id: @storage_id
+  end
+
+  setup do
+    sign_in @project_owner
+  end
+
+  test 'index returns both open and closed code reviews' do
+    script_id = 12
+    level_id = 5
+
+    closed_at = DateTime.now
+    create :code_review, user_id: @project_owner.id, script_id: script_id, level_id: level_id, closed_at: closed_at
+    create :code_review, user_id: @project_owner.id, script_id: script_id, level_id: level_id, closed_at: closed_at + 1.second
+    create :code_review, user_id: @project_owner.id, script_id: script_id, level_id: level_id, closed_at: nil
+
+    get :index, params: {
+      scriptId: script_id,
+      levelId: level_id
+    }
+    assert_response :success
+
+    response_json = JSON.parse(response.body)
+    assert_equal 3, response_json.length
+  end
+
+  test 'create code review' do
+    script_id = 42
+    level_id = 17
+    project_version = 'abc'
+
+    post :create, params: {
+      scriptId: script_id,
+      levelId: level_id,
+      channelId: @channel_id,
+      version: project_version
+    }
+    assert_response :success
+
+    puts response.body
+    response_json = JSON.parse(response.body)
+    assert_not_nil response_json['id']
+    assert_equal script_id, response_json['scriptId']
```

Comment:
Ok, I can move this.  (I actually did look into doing this but it seemed that we generally didn't unit test simple summarize methods?)

Also, I'm curious, after I move these lines to a model test, what (if anything) would you assert here?

Reply:
After thinking about this a bit more, I'd like to keep this verification here because I think it represents parts of the API contract and I want to test as much of the contract in this test as possible.  Given that, do you think a separate test in the code review model is needed?

Reply:
Sounds reasonable to me 👍